File: grouper/WPGConfigViewController.m
//
//  WPGConfigViewController.m
//  grouper
//
//  Created by 高磊 on 14-5-29.
//  Copyright (c) 2014年 WholePort. All rights reserved.
//

#import "WPGConfigViewController.h"
#import "WPGGroupViewController.h"
#import <ArrayUtils.h>

@interface WPGConfigViewController ()

@end

@implementation WPGConfigViewController
{
    NSUInteger perGroup;
    NSUInteger groupCount;
}

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    perGroup = 2;
    groupCount = 4;
    [self updateLabelValue];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)updateLabelValue
{
    self.label_preGroup.text = [NSString stringWithFormat:@"%ld", (long)perGroup];
    self.label_groupCount.text = [NSString stringWithFormat:@"%ld", (long)groupCount];
    self.label_sum.text = [NSString stringWithFormat:@"总人数：%ld", (long)(perGroup * groupCount)];
}

- (IBAction)modify_perGroup:(UIStepper *)sender {
    perGroup = (int)sender.value;
    [self updateLabelValue];
}

- (IBAction)modify_groupCount:(UIStepper *)sender {
    groupCount = (int)sender.value;
    [self updateLabelValue];
}

#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
    if ([[segue destinationViewController] isMemberOfClass:[WPGGroupViewController class]]) {
        
        NSMutableArray *qian = [NSMutableArray array];
        NSArray *words = @[@"A", @"B", @"C", @"D", @"E", @"F", @"G", @"H", @"I", @"J", @"K", @"L", @"M", @"N", @"O", @"P", @"Q", @"R", @"S", @"T", @"U", @"V", @"W", @"X", @"Y", @"Z"];
        for (int i = 0; i <= groupCount - 1; i ++) {
            for (int j = 1; j <= perGroup; j ++) {
                [qian addObject:[words objectAtIndex:i]];
            }
        }
        
        WPGGroupViewController *vc = (WPGGroupViewController *)[segue destinationViewController];
        vc.qian = [qian shuffledArray];
    }
}

@end

File: iGrouping/WPGConfigViewController.swift
//
//  WPGConfigViewController.swift
//  iGrouping
//
//  Created by 高磊 on 15/2/7.
//
//

import UIKit

class WPGConfigViewController: UIViewController {
    
    var perGroup: NSInteger = 0
    var groupCount: NSInteger = 0

    @IBOutlet var label_preGroup: UILabel!
    @IBOutlet var label_groupCount: UILabel!
    @IBOutlet var label_sum: UILabel!
    
    @IBAction func modify_perGroup(sender: UIStepper) {
        perGroup = NSInteger(sender.value)
        updateLabelValue()
    }
    
    @IBAction func modify_groupCount(sender: UIStepper) {
        groupCount = NSInteger(sender.value)
        updateLabelValue()
    }
    
    func updateLabelValue() {
        label_preGroup.text = "\(perGroup)"
        label_groupCount.text = "\(groupCount)"
        let sum: NSInteger = perGroup * groupCount
        label_sum.text = "总人数：\(sum)"
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        perGroup = 2;
        groupCount = 3;
        updateLabelValue()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
        if (segue.destinationViewController.isMemberOfClass(WPGGroupViewController.classForCoder())) {
            let qian: NSMutableArray = NSMutableArray()
            let words: NSArray = NSArray(objects: "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z")
            for i in 0...groupCount-1 {
                for j in 0...perGroup-1 {
                    qian.addObject(words.objectAtIndex(i))
                }
            }
            let vc: WPGGroupViewController = segue.destinationViewController as! WPGGroupViewController
            vc.qian = qian.shuffledArray()
        }
    }

}
